src/main/java/com/github/devotedmc/hiddenore/Config.java

```diff
@@ -47,24 +51,51 @@ private Config() {
 		ignoreSilktouch = false;
 		defaultPrefix = "You found hidden ore!";
 		transformAttemptMultiplier = 3;
+		configurations = new HashMap<String, FileConfiguration>();
 	}
 
 	public static void loadConfig() {
 		try {
-			file = HiddenOre.getPlugin().getConfig();
-			doLoad();
+			mainConfig = HiddenOre.getPlugin().getConfig();
+			doLoadMainConfig(mainConfig);
+			for (World world : HiddenOre.getPlugin().getServer().getWorlds()) {
+				if(world == null) {
+					WorldCreator wc = new WorldCreator(world.getName());
+					HiddenOre.getPlugin().getServer().createWorld(wc);
+				}
+				Reader reader = new InputStreamReader(HiddenOre.getPlugin().getResource("default-world.yml"));
```

Comment:
This whole section is a bit convoluted.

In general I'd recommend rethinking this a bit -- something more direct seems a better approach, namely:

1) Check if the file `%s-config.yml` exists
2) if it does, make a reader, and do `configurations.put(world.getName(), YamlConfiguration.loadConfiguration(worldExistsReader));`
3) if it doesn't, make a reader of internal "default-world.yml" file like you do in this line, then similarly, do a `configurations.put(world.getName(), YamlConfiguration.loadConfiguration(defaultWorldReader))` followed by `configurations.get(world.getName()).save(new File(HiddenOre.getPlugin().getDataFolder(), String.format("%s-config.yml"));`

unless I'm missing something, this will produce the exact same end result, but with a little bit less motion (the file move especially gives me pause).